guides/source/active_support_on_load_hooks.md

```diff
@@ -0,0 +1,105 @@
+
+After reading this guide, you will know:
+
+* What `on_load` hooks are
+* Modifying code to use `on_load` hooks
+* The available hooks inside the Rails framework
+
+--------------------------------------------------------------------------------
+
+What are `on_load` hooks?
+-------------------------
+
+Since Ruby is a dynamic language, some code will cause different Rails frameworks to load. Take this snippet for instance:
+
+```ruby
+ActiveRecord::Base.include(MyActiveRecordHelper)
+```
+
+This snippet means that when this file is loaded, it will encounter `ActiveRecord::Base` right away. This causes Ruby to look for the definition of that constant and then will require it. This causes the entire Active Record framework to be loaded on boot.
+
+`ActiveSupport.on_load` is a mechanism that can be used to defer the loading of code until it is actually loaded. The snippet above can be changed to:
+
+```ruby
+ActiveSupport.on_load(:active_record) { include MyActiveRecordHelper }
+```
+
+This new snippet will only include `MyActiveRecordHelper` when `ActiveRecord::Base` is loaded.
+
+### How does it work?
+
+In the Rails framework these hooks are called when a specific library is loaded. For example, when `ActionController::Base` is loaded, the `:action_controller_base` hook is called. This means that all `ActiveSupport.on_load` calls with `:action_controller_base` hooks will be called in the context of `ActionController::Base` (that means `self` will be an `ActionController::Base`).
+
+Modifying code to use `on_load` hooks
+--------------------------------------
+
+Modifying code is generally straightforward. If you have a line of code that refers to a Rails framework such as `ActiveRecord::Base` you can wrap that code in an `on_load` hook.
+
+### Example 1
+
+```ruby
+ActiveRecord::Base.include(MyActiveRecordHelper)
+```
+
+becomes
+
+```ruby
+ActiveSupport.on_load(:active_record) { include MyActiveRecordHelper } # self refers to ActiveRecord::Base here, so we can simply #include
+```
+
+### Example 2
+
+```ruby
+ActionController::Base.prepend(MyActionControllerHelper)
+```
+
+becomes
+
+```ruby
+ActiveSupport.on_load(:action_controller_base) { prepend MyActionControllerHelper } # self refers to ActionController::Base here, so we can simply #prepend
+```
+
+### Example 3
+
+```ruby
+ActiveRecord::Base.include_root_in_json = true
+```
+
+becomes
+
+```ruby
+ActiveSupport.on_load(:active_record) { self.include_root_in_json = true } # self refers to ActiveRecord::Base here
+```
+
+Available Hooks
+---------------
+
+These are the hooks you can use in your own code.
+
+To mitigate the loading of code, or a conflict on load of one of the classes, use the Available Hooks for that class.
+
+| Class                             | Available Hooks                      |
+| --------------------------------- | ------------------------------------ |
+| `ActiveRecord::Base`              | `active_record`                      |
```

Comment:
You're missing a few: `action_cable`, `action_view_test_case`, `i18n` (not technically a Rails module but a load hook that ActiveSupport runs and that client libraries should use if they need to set a locale or add custom i18n load paths).

You may also want to document here `before_configuration`, `before_initialize`, `before_eager_load`, and `after_initialize`.  They are documented elsewhere, and aren't intended to prevent early loading of components, but they are load hooks nonetheless.

Also, I'd find this much more readable if it were sorted alphabetically, and in particular if classes in the same namespace were grouped consecutively.